sys/mips/timer.c

```diff
@@ -78,12 +75,16 @@ static int mips_timer_start(timer_t *tm, unsigned flags, const bintime_t start,
   assert(flags & TMF_PERIODIC);
   assert(!(flags & TMF_ONESHOT));
 
+  mips32_setcount(0);
+
   device_t *dev = tm->tm_priv;
   mips_timer_state_t *state = dev->state;
   state->sec = 0;
   state->cntr_modulo = 0;
   state->period_cntr = bintime_mul(period, tm->tm_frequency).sec;
   state->compare.val = read_count(state);
+  state->last_count_lo = state->count.lo;
```

Comment:
`read_count()` makes use of `last_count_lo`, so I think you should initialize it to 0 before calling `read_count()` in the line above.